docs/Authenticating-and-Authorizing-Users-via-keycloak.md

```diff
@@ -252,11 +253,98 @@ If all goes well, the following should happen:
 
 If this does not happen, see the Troubleshooting Tips  below.
 
+### Add client for OAuth2 token-based data access
+
+With cBioPortal instances that require user authentication the API can be queried when including a data access token in the request header (see [Authenticating Users via Tokens](Authenticating-Users-via-Tokens.md)). KeyCloak can be configured as an OAuth2 authentication provider that distributes data access tokens to users and validates these tokens when used while querying the API. This feature is enabled by creating a `cbioportal_api` OpenID Connect client that has access to the user roles defined in the `cbioportal` SAML client.
+
+The step below were verified to work with Keycloak versions 4.8.3.Final and 8.0.2.
+
+1. Create a client with name `cbioportal_api`. Set _Client Protocol_ to `openid-connect`.
+
+![](images/previews/oauth2_client_1.png)
+
+2. On the configuration page of `cbioportal_api` client apply the following settings:
+
+#### Settings tab
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| Access Type      | confidential |  |
+| Standard Flow Enabled      | ON      |   (default value) |
+| Enabled      | ON      |   (default value) |
+| Consent Required      | OFF        |   (default value) |
+| Client Protocol      | openid-connect   |   (default value) |
+| Access Type | confidential      |     (default value) |
+| Standard Flow Enabled | ON      |     (default value) |
+| Implicit Flow Enabled | OFF      |     (default value) |
+| Direct Access Grants Enabled | OFF      |    |
+| Service Accounts Enabled | OFF      |  (default value)   |
+| Authorization Enabled | OFF      |  (default value)  |
+| Valid Redirect URIs | _url_/api/data-access-token/oauth2  |  _url_ refers to base url of cBioPortal instance |
+
+![](images/previews/oauth2_client_3.png)
+
+#### Credentials tab
+
+Select `Client Id and Secret`. Take notice of the value of _Secret_ the secret field. This secret should be added to `portal.properties` file of the cBioPortal backend.
```

Comment:
It would be helpful to add a link to line 74 of [Authenticating-Users-via-Tokens.md](https://github.com/cBioPortal/cbioportal/blob/fd3f6e5111ffed07d6eba8975910bdc52743d9e2/docs/Authenticating-Users-via-Tokens.md)

Reply:
Link added.